test/e2e/aws_test.go

```diff
@@ -406,6 +408,42 @@ var _ = Describe("functional tests", func() {
 		})
 	})
 
+	Describe("Cluster should be highly available", func() {
+		It("", func() {
+			By("Creating a workload cluster with single control plane")
+			setup.multipleAZ = true
+			clusterK8sClient := makeSingleControlPlaneCluster(setup)
+
+			By("Creating the second Control Plane Machine in second AZ")
+			awsMachineName := setup.cpAWSMachinePrefix + "-1"
+			bootstrapConfigName := setup.cpBootstrapConfigPrefix + "-1"
+			machineName := setup.cpMachinePrefix + "-1"
+			setup.availabilityZone = availabilityZones[1]
+			createAdditionalControlPlaneMachine(setup, machineName, awsMachineName, bootstrapConfigName)
+
+			By("Creating the MachineDeployment in second AZ")
+			setup.availabilityZone = availabilityZones[1]
+			setup.subnetId = nil
+			createMachineDeployment(setup)
+
+			By("Creating the nginx service exposed as LB")
+			_, ingressHostname := createLBService(metav1.NamespaceDefault, "test-nginx-svc", clusterK8sClient)
```

Comment:
This service is going to be deployed on one of the Nodes that are backing the MachineDeployment, so testing against this service will not give us signal as to the resiliency of the Control Plane as much as the MachineDeployment.